app/script/user/UserRepository.js

```diff
@@ -0,0 +1,925 @@
+      z.util.ko_array_push_all(this.connections, connection_ets);
+
+      // Apply connection to other user entities (which are not us)
+      const user_ids = connection_ets.map((connection_et) => connection_et.to);
+
+      if (user_ids.length) {
+        return this.get_users_by_id(user_ids)
+          .then((user_ets) => {
+            for (const user_et of user_ets) {
+              this._assign_connection(user_et);
+            }
+
+            if (assign_clients) {
+              return this._assign_all_clients()
+                .then(function() {
+                  return resolve();
+                });
+            }
+          });
+      }
+    });
+  }
+
+  /**
+   * Assign all locally stored clients to the users.
+   * @private
+   * @returns {Promise} Promise that resolves with all user entities where client entities have been assigned to.
+   */
+  _assign_all_clients() {
+    return this.client_repository.get_all_clients_from_db()
+      .then((user_client_map) => {
+        this.logger.info(`Found locally stored clients for '${Object.keys(user_client_map).length}' users`, user_client_map);
+        const user_ids = [];
+
+        for (const user_id in user_client_map) {
+          user_ids.push(user_id);
+        }
+
+        return this.get_users_by_id(user_ids)
+          .then((user_ets) => {
+            for (const user_et of user_ets) {
+              if (user_client_map[user_et.id].length > 8) {
+                this.logger.warn(`Found '${user_client_map[user_et.id].length}' clients for '${user_et.name()}'`, user_client_map[user_et.id]);
+              }
+              user_et.devices(user_client_map[user_et.id]);
+            }
+
+            return user_ets;
+          });
+      });
+  }
+
+  /**
+   * Assign connections to the users.
+   * @param {z.entity.User} user_et - User to which a connection will be assigned to.
+   * @returns {Promise} Promise that resolves with the connection entity which has been found for the given user entity.
+   * @private
+   */
+  _assign_connection(user_et) {
+    const connection_et = this.get_connection_by_user_id(user_et.id);
+    if (connection_et) {
+      user_et.connection(connection_et);
+    }
+    return connection_et;
+  }
+
+  /**
+   * Update the status of a connection.
+   * @private
+   * @param {z.entity.User} user_et - User to update connection with
+   * @param {string} status - Connection status
+   * @param {boolean} show_conversation - Show conversation on success
+   * @returns {Promise} Promise that resolves when the connection status was updated
+   */
+  _update_connection_status(user_et, status, show_conversation = false) {
+    return Promise.resolve()
+      .then(() => {
+        if (user_et.connection().status() === status) {
+          return;
+        }
+        return this.user_service.update_connection_status(user_et.id, status);
+      })
+      .then((response) => {
+        return this.user_connection(response, show_conversation);
```

Comment:
This would try to map a response of undefined. Probably fails. Did we previously do the same?